doc/source/train/user-guides/monitoring-logging.rst

```diff
@@ -77,6 +32,38 @@ metrics from multiple workers.
 
         Here is an example of reporting both the aggregated R2 score and mean train and validation loss from all workers.
 
-        .. literalinclude:: ../doc_code/torchmetrics_example.py
+        .. literalinclude:: ../doc_code/metric_logging.py
             :language: python
-            :start-after: __start__
+            :start-after: __torchmetrics_start__
+            :end-before: __torchmetrics_end__
+
+
+.. _train-metric-only-reporting-deprecation:
+
+(Deprecated) Reporting free-floating metrics
+--------------------------------------------
+
+Reporting metrics with ``ray.train.report(metrics, checkpoint=None)`` from every worker writes the metrics to a Ray Tune log file (``progress.csv``, ``result.json``)
+and is accessible via the ``Result.metrics_dataframe`` on the :class:`~ray.train.Result` returned by ``trainer.fit()``.
+
+As of Ray 2.43, this behavior is deprecated and will not be supported in Ray Train V2,
+which is an overhaul of Ray Train's implementation and select APIs.
+
+Ray Train V2 only keeps a slim set of experiment tracking features that are necessary for fault tolerance, so it does not support reporting free-floating metrics that are not attached to checkpoints.
+The recommendation for metric tracking is to report metrics directly from the workers to experiment tracking tools such as MLFlow and WandB.
+See :ref:`train-experiment-tracking-native` for examples.
+
+In Ray Train V2, reporting only metrics from all workers is a no-op. However, it is still possible to access the results reported by all workers to implement custom metric-handling logic.
+
+.. literalinclude:: ../doc_code/metric_logging.py
+    :language: python
+    :start-after: __report_callback_start__
+    :end-before: __report_callback_end__
+
+
+To use Ray Tune :class:`Callback <ray.tune.Callback>` that depend on free-floating metrics reported by workers, :ref:`run Ray Train as a single Ray Tune trial. <train-with-tune-callbacks>`
```

Comment:
nit (though not sure if this syntax works, maybe need to move the s into ``:class:`Callbacks <ray.tune.Callback>` ``)

```suggestion
To use Ray Tune :class:`Callback <ray.tune.Callback>`s that depend on free-floating metrics reported by workers, :ref:`run Ray Train as a single Ray Tune trial. <train-with-tune-callbacks>`
```